contrib/ansible/roles/docker/tasks/main.yml

```diff
@@ -0,0 +1,30 @@
+---
+- name: Create systemd conf drop
+  file: dest=/usr/lib/systemd/system/docker.service.d state=directory
+  tags: docker
+
+- name: Install http_proxy into docker drop-in
+  copy: dest=/usr/lib/systemd/system/docker.service.d/51-http-proxy.conf content="[Service]\nEnvironment="HTTP_PROXY={{ http_proxy }}""
+  when: http_proxy is defined
+  notify: 
+    - restart docker
+
+- name: Install https_proxy into docker drop-in
+  copy: dest=/usr/lib/systemd/system/docker.service.d/52-https-proxy.conf content="[Service]\nEnvironment="HTTPS_PROXY={{ https_proxy }}""
+  when: https_proxy is defined
+  notify: 
+    - restart docker
+
+- name: Install no-proxy into docker drop-in
+  copy: dest=/usr/lib/systemd/system/docker.service.d/49-no-proxy.conf content="[Service]\nEnvironment="NO_PROXY={{ no_proxy }}""
+  notify: 
+    - restart docker
```

Comment:
To be honest, I don't understand what is happening here vs using /etc/sysconfig/docker like you do below...